sdk/core/azure-core/azure/core/pipeline/transport/_pyodide.py

```diff
@@ -0,0 +1,150 @@
+# --------------------------------------------------------------------------
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# The MIT License (MIT)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the ""Software""), to
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+
+from collections.abc import AsyncIterator
+from io import BytesIO
+
+from pyodide import JsException  # pylint: disable=import-error
+from pyodide.http import pyfetch  # pylint: disable=import-error
+
+from azure.core.utils import CaseInsensitiveDict
+from azure.core.exceptions import HttpResponseError
+from azure.core.rest._http_response_impl_async import AsyncHttpResponseImpl
```

Comment:
Is this a rest transport or a pipeline transport?

Reply:
I didn't know there was a difference. I will investigate

Reply:
spoke to @iscai-msft. This is a rest transport.

Reply:
Then you may want to put it under azure/core/rest?

Reply:
@xiangyan99 `azure.core.rest` is just for exposing `HttpRequest` and `HttpResponse`, we still want to keep transports under `azure.core.pipeline.transport`

Reply:
My understanding is 

class PyodideTransportResponse(AsyncHttpResponseImpl)

should be under azure/core/rest?

Reply:
we explicitly don't expose transport-specific responses. That's only for internal use, we only expose `azure.core.rest.HttpResponse` and `azure.core.rest.AsyncHttpResponse`, which are protocols for the responses. The transport-specific responses are for us to make the response returned from the transport conform to the public response protocol